hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/TransactionUtils.java

```diff
@@ -137,4 +165,20 @@
       throw new HoodieIOException("Unable to read metadata for instant " + hoodieInstantOption.get(), io);
     }
   }
+
+  /**
+   * Get pending clustering instant.
+   * Notice:
+   *   we return .requested instant here.
+   *
+   * @param metaClient
+   * @return
+   */
+  public static List<HoodieInstant> getPendingReplaceRequestedInstants(HoodieTableMetaClient metaClient) {
+    return metaClient
+        .getActiveTimeline()
+        .filterPendingReplaceTimeline()
+        .getInstants()
+        .map(f -> HoodieTimeline.getReplaceCommitRequestedInstant(f.getTimestamp())).collect(Collectors.toList());
```

Comment:
I understand this addresses the completed replacecommit which was pending before the lastSuccessfulCommit looked by the conflict resolution.  Is there any other type of pending instants we need to consider as well, e.g., commits, deltacommits, compaction, etc.?  They could also be missed during conflict resolution like the pending clustering?